core/domain/email_manager.py

```diff
@@ -246,3 +261,44 @@ def send_moderator_action_email(
     _send_email(
         recipient_id, sender_id, intent, email_subject, full_email_content,
         bcc_admin=True)
+
+def _generate_hash(
+        recipient_id, email_subject, email_html_body):
+    """Generate hash for a given recipient_id, email_subject
+        and cleaned email_html_body.
+    """
+
+    hash_value = utils.convert_to_hash(
+        recipient_id + email_subject + email_html_body,
+        100)
+
+    return hash_value
+
+def _check_duplicate_message(
+        recipient_id, email_subject, email_html_body):
+    """Check for a given recipient_id, email_subject
+        and cleaned email_html_body, whether a similar message has been sent
+        in the last DUPLICATE_EMAIL_INTERVAL_MINS.
+    """
+
+    email_hash = _generate_hash(recipient_id, email_subject, email_html_body)
+
+    after = datetime.datetime.utcnow() - datetime.timedelta(
+        minutes=feconf.DUPLICATE_EMAIL_INTERVAL_MINS)
+
+    messages = email_models.SentEmailModel.get_by_hash(email_hash, after)
```

Comment:
Style nit: this should be get_by_hash(email_hash, sent_datetime_lower_bound=after)

(for args with default values, we write the arg name explicitly when calling)

Ditto elsewhere.